schema/field/field.go

```diff
@@ -208,6 +208,12 @@ func (b *stringBuilder) Default(s string) *stringBuilder {
 	return b
 }
 
+// DefaultFunc sets the default value generating function of the field.
+func (b *stringBuilder) DefaultFunc(fn func() string) *stringBuilder {
+	b.desc.Default = fn()
+	return b
+}
```

Comment:
The function needs to be invoked by the creator and not by the schema.
```suggestion
func (b *stringBuilder) DefaultFunc(fn func() string) *stringBuilder {
	b.desc.Default = fn
	return b
}
```

Start with this change and I can help you to add it to the codegen templates.

Reply:
done

Reply:
cc @a8m